src/extension.ts

```diff
@@ -255,13 +250,17 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
     const phanScriptPathValidated = phanScriptPath;  // work around typescript complaint.
 
     // Check if the analyzedProjectDirectories setting was provided.
-    if (!analyzedProjectDirectories) {
-        if (originalAnalyzedProjectDirectories instanceof Array) {
-            await showOpenSettingsPrompt('The setting phan.analyzedProjectDirectories must be a path or a non-empty array of paths (e.g. /path/to/some/project_folder). `.phan` must be a folder within that directory.');
-        } else {
-            await showOpenSettingsPrompt('The setting phan.analyzedProjectDirectories must be provided (e.g. /path/to/some/project_folder). `.phan` must be a folder within that directory.');
+    if (!analyzedProjectDirectories.length) {
+        analyzedProjectDirectories = findValidProjectDirectories();
+
+        if (!analyzedProjectDirectories.length) {
+            // Do not send an error to the interface, this is frustrating.
```

Comment:
It's also frustrating to install an extension for a tool you're unfamiliar with and have it not work (e.g. some users may try to start using the extension before setting up a Phan config for the project).

I'd rather you added a config setting such as `phan.silenceNoDirectoryWarning: bool`

Reply:
Yea, sounds good, but also think about this: when you install an extension, you go to the vscode extension page. There, one of the first thing a user must see is this: 

"For this extension to work, a workspace folder must contain the ".phan" folder with a"config.php" file inside. To add custom folders and override auto-detection use the setting "analizedProjectDirectories" ".

Having a bunch of configs for every control structure I don't think it is a good idea. An extension should be pleasant and simple to use. Additionally we can alert an error when the setting to debug is on, this is when an user could not figure out how to use, and enabling this will throw the error.

Also I think analizedProjectDirectories should be also make to detect relative paths, but that's another story.